src/averageAges.js

```diff
@@ -37,7 +42,18 @@ function calculateMenAverageAge(people, century) {
  * @return {number}
  */
 function calculateWomenAverageAge(people, withChildren) {
-  // write code here
+  let women = people.filter(person => person.sex === 'f');
+
+  if (withChildren) {
+    women = women.filter(
+      woman => people.some(child => child.mother === woman.name)
+    );
+  }
```

Comment:
```suggestion
  if (withChildren) {
    women = women.filter(
      woman => people.some(child => child.mother === woman.name)
    );
  }
```

eplace `if ()` statement with &&, || or ?: